lib/gui/lvgl/lvgl.c

```diff
@@ -77,18 +77,28 @@ static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_drv)
 	struct display_capabilities cap;
 	const struct device *display_dev = (const struct device *)disp_drv->user_data;
 	int err = 0;
+	uint16_t rows, cols;
 
 	display_get_capabilities(display_dev, &cap);
 
-	if (cap.x_resolution <= CONFIG_LVGL_HOR_RES_MAX) {
-		disp_drv->hor_res = cap.x_resolution;
+	if (cap.current_orientation == DISPLAY_ORIENTATION_NORMAL ||
```

Comment:
This code gives me more of a feeling that there is a bug in a driver.
The driver should always report X direction as horizontal pixels and Y direction as vertical coordinates even if the display gets rotated.

Reply:
Display API does not clarify the behavior. If a display reports it is rotated 90,270, should it also swap its X,Y resolution or not? If display drivers swap X,Y, we need then to handle that on the kscan rotation equations. It's actually the solution implemented in the previous commit: fb32552fb47972d74d74d8256ece398d48b55639.

Reply:
True that the API does not specify it on the capabilities struct but the read/write functions of the API specify that the x,y coordinates are seen from the upper left corner.